src/hydra/dataProvider.ts

```diff
@@ -604,6 +609,51 @@ function dataProvider(
         ),
       );
 
+  /*
+   * The fetchAllPages method allows running as many requests as needed in order to load all pages of a list.
+   * This function uses the already transformed react-admin response data and could be improved by using the hydra pagination
+   */
+  const fetchAllPages = async (
+    resource: string,
+    params: Partial<GetListParams>,
+    previousResult?: GetListResult,
+  ): Promise<GetManyResult> => {
+    params.pagination = {
+      // Using a default perPage value big enough to limit the number of requests
+      // If the API has a max_per_page value, the perPage parameter can be ignored
+      perPage: params.pagination?.perPage || 100,
```

Comment:
I don't think we should set a default value when there is no `perPage` parameter.

Reply:
If you look at the `GetListParams` type, you can see that we must provide both perPage & page parameters with a number: they are not optional for react-admin.

Indeed, the hydra provider does not requires them for now, but I guess you are going to type everything with react-admin types soon.

Reply:
Alright.